### DISCUSSION TITLE
Deleting error  P2003

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [prisma/prisma#21093] [Q&A] [answered]
### Question

I have a table user where a user can have a lot of posts but post belong to one user its a many to one relationship I will leave to you my schema.prisma.
 this is my controller : 
```ts
 @Delete('id')
    remove(@Param('id') id : string){
        this.commentService.remove(+id)
    }
```
and my service : 
```ts
 remove(id: number) {
    
    return this.prisma.user.delete({where : {id}} )
  }
```

Thank you in advance :) 

### Information about Prisma Schema, Client Queries and Environment (optional)
```prisma
model user{
  id Int @id @default(autoincrement())
  firstname String 
  lastname String
  username String @unique
  email String @unique
  password String
  imageUrl String?

  posts post[]
  comments comment[]

}

model post{
  id Int @id @default(autoincrement())
  content String
  imagePost String?
  createdDate  DateTime @default(now())
  
  author user @relation(fields: [authorId],references: [id])
  authorId Int

  comments comment[]
  
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @Elmatbaisaad 👋 

Thank you for raising this question.

When you try to delete a `user`, there might be `posts` with a foreign key that references a `user`. Because of this relation, you will get this error. You can use [referential actions](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions). In your case, add `OnDelete: Cascade` inside the relation field in the `user` model.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by Elmatbaisaad** [comment]
Thank you for your answer I did it like that : 
```prisma
model user{
  id Int @id @default(autoincrement())
  firstname String 
  lastname String
  username String @unique
  email String @unique
  password String
  imageUrl String?
  posts post[]
  comments comment[]

}

model post{
  id Int @id @default(autoincrement())
  content String
  imagePost String?
  createdDate  DateTime @default(now())
  
  author user @relation(fields: [authorId],references: [id], onDelete: Cascade)
  authorId Int

  comments comment[]
  
} 
```
I got that err 
 
<img width="1080" alt="Capture d’écran 2023-09-15 à 12 08 25" src="https://github.com/prisma/prisma/assets/101296742/4b224bd3-23e8-452b-a451-ab4defd109e5">

**Reply by ludralph** [reply]
This error occurs when the record with the specified `id` does not match any records in the database. Are you sure that the record you are trying to delete exists in the database?

**Comment by Elmatbaisaad** [comment]
![Capture d’écran 2023-09-15 à 13 52 22](https://github.com/prisma/prisma/assets/101296742/6caea979-5a98-4983-8c56-b04ac02cf0da)
Yess
